src/box/alter.cc

```diff
@@ -1739,6 +1739,14 @@ alter_space_move_indexes(struct alter_space *alter, uint32_t begin,
 	} else {
 		is_min_field_count_changed = false;
 	}
+	/*
+	 * When space upgrade completes, we should unconditionally update
+	 * optionality because current min_field_count value of a format
+	 * respects min_field_count of format which space had before upgrade.
+	 * It was needed to correctly index both old and new tuples.
+	 */
+	if (alter->old_space->def->opts.upgrade_def != NULL)
+		is_min_field_count_changed = true;
```

Comment:
Nit:
- You can use `old_space` instead of `alter->old_space` here.
- I'd put in one `if-else-if-else` block.

```C
if (old_space->def->opts.upgrade_def != NULL) {
	/* ... */
	is_min_field_count_changed = true;
} else if (old_space->format != NULL) {
	is_min_field_count_changed =
		old_space->format->min_field_count !=
		alter->new_min_field_count;
} else {
	is_min_field_count_changed = false;
}
```

Reply:
Fixed.